core/types/receipt.go

```diff
@@ -600,13 +600,16 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 	signer := MakeSigner(config, new(big.Int).SetUint64(number), time)
 
 	logIndex := uint(0)
-	if len(txs) != len(rs) {
+
+	// If rs are one longer than txs it indicates the presence of a celo block receipt.
+	if len(txs) != len(rs) && len(txs)+1 != len(rs) {
```

Comment:
In the old code we also check the hash of the receipt: https://github.com/celo-org/celo-blockchain/blob/e67c1fd73f4e4441bc55a5b3793ba11549a53d2d/core/rawdb/accessors_chain.go#L703

Reply:
That code was broken, it was checking the receiptBlock hash matched the receiptTx hash, but those are only set further down in DeriveFields. More detailed description here - https://github.com/celo-org/celo-blockchain/issues/2288